### ISSUE TITLE
How does CatBoost gradient boosting work in multiclass mode ?

### ISSUE THREAD

**Issue by YohannLeFaou** [catboost/catboost#1806]
Section 4.6 in Friedman (https://projecteuclid.org/journals/annals-of-statistics/volume-29/issue-5/Greedy-function-approximation-A-gradient-bostingmachine/10.1214/aos/1013203451.full) explains how to compute gradients based on a crossentropy loss for a K classes classification problem, and then suggests to build K trees for each boosting iteration.

However, when looking at the CatBoost model (with a json export) it seems that only one tree is built for each boosting iteration.

I found this documentation about scoring fucntions in your website: https://catboost.ai/docs/concepts/algorithm-score-functions.html but I would like to know what you are doind exactly for the multiclass case.

Thanks.

**Comment by kizill** [comment]
We have a special loss function 
![image](https://user-images.githubusercontent.com/682769/129784374-3ff077f3-e4ca-4301-ae04-777ec60039c3.png)
that uses multidimensional predictions for each object. In each leaf of a newly built tree we store M predictions for each of M classes. For that function we are able to calculate first and second derivatives and then we can calculate regular cosine or l2 distance between "ideal" approxes and mean gradient in leaf, just as in any other loss function.
And multiclass one-vs-all mode works in the same manner - single class models share the same tree structures.
https://catboost.ai/docs/concepts/loss-functions-multiclassification.html

**Comment by YohannLeFaou** [comment]
Ok I got it finally. Thank you for your answer.
